src/main/java/techreborn/blockentity/machine/multiblock/BlastFurnaceRecipeCrafter.java

```diff
@@ -0,0 +1,114 @@
+package techreborn.blockentity.machine.multiblock;
+
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.item.ItemStack;
+import reborncore.common.crafting.RebornRecipe;
+import reborncore.common.recipes.RecipeCrafter;
+import reborncore.common.util.RebornInventory;
+import techreborn.init.ModRecipes;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BlastFurnaceRecipeCrafter extends RecipeCrafter {
+
+	private int ticksSinceLastChange;
+
+	public BlastFurnaceRecipeCrafter(BlockEntity blockEntity, int inputs, int outputs, RebornInventory<?> inventory,
+									 int[] inputSlots, int[] outputSlots){
+		super(ModRecipes.BLAST_FURNACE,blockEntity,inputs, outputs, inventory, inputSlots, outputSlots);
+
+	}
+
+	private List<ItemStack> getCurrentInputs(){
+		List<ItemStack> inputStacks = new ArrayList<>();
+		for (int inputSlot : inputSlots) {
+			inputStacks.add(inventory.getStack(inputSlot));
+		}
+		return inputStacks;
+	}
+
+	private List<ItemStack> getPartialOutput(RebornRecipe recipe, List<ItemStack> inputs){
+		for(ItemStack input : inputs){
+			if(input.isDamageable()){
+				float durabilityPercentage = 1 - ((float)input.getDamage()) / input.getMaxDamage();
+				return recipe.getOutputs().stream().map(itemStack -> {
+					ItemStack copied = itemStack.copy();
+					if (copied.getCount() > 1) {
+						copied.setCount((int) Math.floor(copied.getCount() * durabilityPercentage));
+					}
+					return copied;
+				}).toList();
+			}
+		}
+		return recipe.getOutputs();
+	}
+
+	@Override
+	public void updateEntity() {
+		if (blockEntity.getWorld() == null || blockEntity.getWorld().isClient) {
+			return;
+		}
+		ticksSinceLastChange++;
+		// Force a has chanced every second
+		if (ticksSinceLastChange == 20) {
+			setInvDirty(true);
+			ticksSinceLastChange = 0;
+			setIsActive();
+		}
+		// It will now look for new recipes.
+		if (currentRecipe == null && isInvDirty()) {
+			updateCurrentRecipe();
+		}
+		if (currentRecipe != null) {
+			// If it doesn't have all the inputs reset
+			if (isInvDirty() && !hasAllInputs()) {
+				currentRecipe = null;
+				currentTickTime = 0;
+				setIsActive();
+			}
+			// If it has reached the recipe tick time
+			if (currentRecipe != null && currentTickTime >= currentNeededTicks && hasAllInputs()) {
+				boolean canGiveInvAll = true;
+				List<ItemStack> recipeOutput = getPartialOutput(currentRecipe, getCurrentInputs());
```

Comment:
Wouldn't it be better to override getOutputs method of BlastFurnaceRecipe instead of additional crafter?

Reply:
I tried that first but i could not find a way to get the input itemstack damage from that context. I will try again. Thank you

Reply:
Upon further investigation. I sadly could not find a nice way to determine the amount of returned items in the getOutputs method.